src/core/Syntax.jl

```diff
@@ -406,6 +406,90 @@ function functor(types::Tuple, expr::GATExpr;
   invoke_term(theory_type, types, name, term_args...)
 end
 
+abstract type AbstractTheoryMap end
+
+function fmap(T::AbstractTheoryMap, x::Expr) end
```

Comment:
Should be `x::GATExpr` instead of `x::Expr`. That's probably why you're getting the error.

Reply:
I changed it, and it didn't seem to fix the error.